File: Poo_java/tp2/tp2_ex2/src/Ppoint.java
public class Ppoint {
    private int x;
    private int y;

    public Ppoint(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getY() {
        return this.y;
    }

    public int getX() {
        return this.x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    public void afficher() {
        System.out.println(" abs = " + x);
        System.out.println(" ord = " + y);
    }

    public Ppoint increment() {
        this.x++;
        this.y++;
        return this;
    }

    public Ppoint decrement() {
        this.x--;
        this.y--;
        return this;
    }
}

File: TpPoo_cpp/tp2_ex2/Ppoint.cpp
//
// Created by amani on 01/03/24.
//

#include "Ppoint.h"
Ppoint::Ppoint(int x, int y) {
    this->x=x;
    this->y=y;
}
int Ppoint::getY() {
    return this->y;
}
int Ppoint::getX() {
    return this->x;
}
void Ppoint::setX(int x) {
    this->x=x;
}
void Ppoint::setY(int y) {
    this->y=y;
}

void Ppoint::afficher(){
    cout<<" abs = "<<x<<endl;
    cout<<" ord = "<<y<<endl;
};
Ppoint operator ++(Ppoint& p){
    p.x++;
    p.y++;
    return p;
};
Ppoint operator --(Ppoint &p){
    p.x--;
    p.y--;
    return p;
};
